sycl/source/detail/event_impl.cpp

```diff
@@ -183,23 +183,9 @@ event_impl::event_impl(queue_impl &Queue, private_tag)
 }
 
 event_impl::event_impl(HostEventState State, private_tag) : MState(State) {
-  switch (State) {
-  case HES_Discarded:
-  case HES_Complete: {
+  MIsHostEvent = true;
```

Comment:
Looks like this wasn't set for incomplete events before this PR. Do we have a bunch of `= true` elsewhere that we can remove now?